src/fastapi_app/api_routes.py

```diff
@@ -52,7 +54,7 @@ async def search_handler(query: str, top: int = 5, enable_vector_search: bool =
     return [item.to_dict() for item in results]
 
 
-@router.post("/chat")
+@router.post("/chat", response_model=RetrievalResponse)
```

Comment:
This makes swagger API detect Schema for response and cast it according to this type.